### PR TITLE
Bootstrap ssot integration

### PR DESCRIPTION
**Pull request by bile0026**
Add Bootstrap SSoT to sync in information from a local yaml file or git repository hosting information for various Nautobot "base" models.

### CODE REVIEW

**Comment by Kircheneer** [comment]
Looks much better already

**Review by jdrew82 — CHANGES_REQUESTED** [review]
Just some items to address.

File: docs/admin/integrations/bootstrap_setup.md [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+        },
+      }
+}
+```
+
+## Configuration
+
+### Bootstrap data
+
+Bootstrap data can be stored in 2 fashions. Firstly, it can be stored within the `nautobot_ssot_bootstrap/fixtures` directory, or you may create a Git Repository within an existing Nautobot instance that contains the word `Bootstrap` in the name and provides `config context` data. The data structure is flat files, there is a naming scheme to these files. The first one required is `global_settings.yml`. This contains the main data structures of what data can be loaded `Secrets,SecretsGroups,GitRepository,DynamicGroup,Tag,etc`. You can then create additional `.yml` files with naming of your CI environments, i.e. production, development, etc. This is where the environment variables described below would be matched to pull in additional data from the other yaml files defined in the directory. A simple structure would look something like this:
```
[the remark this answers is not in the document]
Comment (bile0026) [reply docs/admin/integrations/bootstrap_setup.md:59]: Made some clarification here

File: nautobot_ssot/integrations/bootstrap/diffsync/adapters/nautobot.py [hunk 2]
```diff
@@ -0,0 +1,1327 @@
...
+            sync (object, optional): Nautobot DiffSync. Defaults to None.
+        """
+        super().__init__(*args, **kwargs)
+        self.job = job
+        self.sync = sync
+
+    def load_tenant_group(self):
+        """Method to load TenantGroup objects from Nautobot into NautobotTenantGroup DiffSync models."""
+        for nb_tenant_group in TenantGroup.objects.all():
+            self.job.logger.debug(f"Loading Nautobot TenantGroup: {nb_tenant_group}, with ID: {nb_tenant_group.id}")
```
[the remark this answers is not in the document]
Comment (bile0026) [reply nautobot_ssot/integrations/bootstrap/diffsync/adapters/nautobot.py:197]: fixed.

File: docs/admin/integrations/bootstrap_setup.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (jdrew82) [reply docs/admin/integrations/bootstrap_setup.md]: I'm not seeing any clarification around the fixtures folder usage that you're suggesting. I think we still need clarification on what you're expecting users to do in that particular use case.

File: docs/admin/integrations/bootstrap_setup.md [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+staging.yml
+```
+
+There are 2 environment variables that control how certain things are loaded in the app.
+
+  1. `NAUTOBOT_BOOTSTRAP_SSOT_LOAD_SOURCE` - defines whether to load from the local `fixtures` folder or a GitRepository already present in Nautobot.
+    - Acceptable options are `file` or `git`.
+  2. `NAUTOBOT_BOOTSTRAP_SSOT_ENVIRONMENT_BRANCH` - Defines the environment and settings you want to import. I.e. production, develop, staging.
+
+## Process
```
[the remark this answers is not in the document]
Comment (jdrew82) [reply docs/admin/integrations/bootstrap_setup.md:80]: The Process section is specifically talking about how you use the App and is not talking about installation.

File: nautobot_ssot/integrations/bootstrap/constants.py [hunk 4]
```diff
@@ -0,0 +1,3 @@
+"""Constants to be used with the nautobot_ssot_bootstrap plugin."""
```
[the remark this answers is not in the document]
Comment (jdrew82) [reply nautobot_ssot/integrations/bootstrap/constants.py]: OK, so delete this if it's not needed please.

**Review by jdrew82 — COMMENTED** [review]
Still some stuff to address.

File: docs/user/integrations/index.md [hunk 5]
```diff
@@ -3,7 +3,9 @@
 This Nautobot app supports the following integrations:
 
 - [Cisco ACI](./aci.md)
+- [Bootstrap](./bootstrap.md)
 - [Arista CloudVision](./aristacv.md)
+- [Bootstrap](./bootstrap.md)
```
Comment (chadell) [inline docs/user/integrations/index.md:8]: why twice?
```suggestion
```

Comment (bile0026) [reply docs/user/integrations/index.md:8]: fixed

File: docs/admin/integrations/bootstrap_setup.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (bile0026) [reply docs/admin/integrations/bootstrap_setup.md:80]: This is part of the configuration section?

[the remark this answers is not in the document]
Comment (jdrew82) [reply docs/admin/integrations/bootstrap_setup.md:80]: This is in the admin section which is for installation instructions. The process section that you have her is about USING the integration so should be placed in the User documentation section and should have some screenshots of using the integration.

File: docs/admin/integrations/bootstrap_setup.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (bile0026) [reply docs/admin/integrations/bootstrap_setup.md]: Adjusted this as well. Let me know if this makes more sense.

File: docs/admin/integrations/bootstrap_setup.md [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+}
+```
+
+## Configuration
+
+### Bootstrap data
+
+Bootstrap data can be stored in 2 fashions.
+
+1. (Reccomended) Bootstrap data can be stored in a Git Repository and referenced in the app as a Git Datasource. A user should create a Git Repository in Nautobot (including any necessary Secrets and SecretsGroups for access) with the word "Bootstrap" in the name, and with a provided content type of `config contexts`. This is how the App will locate the correct repository. The data structure is flat files, and there is a naming scheme to these files. The first one required is `global_settings.yml`. This contains the main data structures of what data can be loaded `Secrets,SecretsGroups,GitRepository,DynamicGroup,Tag,etc`. You can then create additional `.yml` files with naming of your CI environments, i.e. production, development, etc for default values for specific items. This is where the environment variables described below would be matched to pull in additional data from the other YAML files defined in the directory.
```
Comment (gsnider2195) [inline docs/admin/integrations/bootstrap_setup.md:67]:
```suggestion
1. (Recommended) Bootstrap data can be stored in a Git Repository and referenced in the app as a Git Datasource. A user should create a Git Repository in Nautobot (including any necessary Secrets and SecretsGroups for access) with the word "Bootstrap" in the name, and with a provided content type of `config contexts`. This is how the App will locate the correct repository. The data structure is flat files, and there is a naming scheme to these files. The first one required is `global_settings.yml`. This contains the main data structures of what data can be loaded `Secrets,SecretsGroups,GitRepository,DynamicGroup,Tag,etc`. You can then create additional `.yml` files with naming of your CI environments, i.e. production, development, etc for default values for specific items. This is where the environment variables described below would be matched to pull in additional data from the other YAML files defined in the directory.
```

**Review by jdrew82 — CHANGES_REQUESTED** [review]
One small nitpick to the docs. Also, I think you need to add yourself to the CODEOWNERS for this integration.

**Comment by jdrew82** [comment]
Also, please fix conflict with release notes for 1.5.

File: docs/admin/integrations/bootstrap_setup.md [hunk 7]
```diff
@@ -0,0 +1,84 @@
...
+
+## Installation
+
+Add the Nautobot SSoT App to your poetry environment `poetry install nautobot-ssot`, then configure your `nautobot_config.py` to include the app and the settings.
```
[the remark this answers is not in the document]
Comment (bile0026) [reply docs/admin/integrations/bootstrap_setup.md:10]: Followed device42 example

**Review by jdrew82 — APPROVED** [review]
LGTM!

**Pull request merged**